src/proto/TheOpenNetwork.proto

```diff
@@ -67,6 +67,9 @@ message JettonTransfer {
 
     // Amount in nanotons to forward to recipient. Basically minimum amount - 1 nanoton should be used
     uint64 forward_amount = 5;
+
+    // Optional payload to attach to the jetton transfer message
+    CustomPayload custom_payload = 6;
```

Comment:
The things are getting more complicated. Do you mind to refactor the Protobuf API to make it better for different custom payload support?
1. Move `CustomPayload.state_init` to `Transfer.state_init`
2. Change type of `Transfer.payload_oneof.custom_payload` variant from `CustomPayload` to `bytes`
3. Change type of `JettonTransfer.custom_payload` from `CustomPayload` to `bytes`

It will make it more clear, easier to maintain and even use. Otherwise we'll have a mess with the state inits in `Transfer` and `JettonTransfer` messages

Reply:
I totally agree with you. I plan to change it to:

<img width="1354" alt="image" src="https://github.com/user-attachments/assets/03daf568-e695-4041-9e32-ec7a86c6b16d">

Just a reminder that this is a breaking change.

There's one more detail: we don't need to change it to the bytes type; we can continue using the previous string type, right? This is because the RPC returns base64 strings, which are also convenient for users.

Reply:
The changes look exactly as I meant 👍 

It should be ok as the change will make the overall UX much better
> Just a reminder that this is a breaking change.

Yes, right, I just made a mistake when wrote the comment. It should be string as is now
> There's one more detail: we don't need to change it to the bytes type; we can continue using the previous string type, right?

Reply:
The code has been adjusted according to the comments, but I wasn't able to successfully run the Android/iOS test cases due to issues with my local environment.